client.go

```diff
@@ -90,11 +92,14 @@ func (h *connectionHandler) newConnection(path string) error {
 // connect tries to connect to the service with retries given the configuration
 // strategy
 func (h *connectionHandler) connect(dns string, maxRetries int, bckoff string) error {
+	// This block is executed in the handlers to avoid launching too many goroutines
+	// that will do nothing because of the atomic.Bool set to true
+	// if !h.reconnecting.CompareAndSwap(false, true) {
+	// 	return fmt.Errorf("already reconnecting")
+	// }
```

Comment:
I would add a "warning", just one line before the actual comment:
```
WARNING: this function should not be called before setting the reconnecting flag to true:
```

just because I often overlook comments :D